### ISSUE TITLE
Plaidml stuck on last sample for some time of each epoch 

### ISSUE THREAD

**Issue by Devesh24** [plaidml/plaidml#345]
Hi Guys 

**Platform**: Ubuntu 18.04, AMD R9 390
**Problem** : Whenever i try to train any CNN model created on Keras , it stuck at last sample of each 
epoch for 3-4 minutes, i am using dogs-cats data set with following params
img_width , img_height = 128, 128
batch_size = 64
epochs = 50

**Logs** :    Epoch 1/50
359/359 [==============================] - 391s 1s/step - loss: 0.6467 - acc: 0.6115 - val_loss: 0.6120 - val_acc: 0.6680
Epoch 2/50
358/359 [============================>.] - ETA: 0s - loss: 0.6028 - acc: 0.6622

**Comment by tzerrell** [comment]
Keras runs validation at the end of each epoch without displaying a progress bar; this behavior occurs when validation takes a long time. If you are seeing this behavior on networks that do not have the same problem with other backends (e.g. tensorflow) then the underlying problem is our performance. Is this a network you've also run with other backends, and if so, does that make this issue go away?

If you are running this only with the PlaidML backend, check if your validation set is appropriately sized. If it has bad relative performance compared to the training portion (e.g. if the amount of validation you do is smaller than the amount of training you do per epoch, but validation takes longer), that is a surprising performance issue and we're particularly interested in more details.

**Comment by Devesh24** [comment]
Hi @tzerrell 
I have tried same code on google colab GPU platform and it is giving same issue, i think you are right there about validation running in background. I think this problem is related to https://github.com/keras-team/keras/issues/6101.

Thanks a lot for help.
